lib/rubygems/compiler.rb

```diff
@@ -68,6 +68,11 @@ def adjust_gemspec_files(gemspec, artifacts)
       gemspec.files.reject! { |f| !File.exist?("#{target_dir}/#{f}") }
     end
 
+    if @options[:sign_private_key_file]
+      gemspec.cert_chain = [@options[:sign_cert_file]]
+      gemspec.signing_key = @options[:sign_private_key_file]
+    end
```

Comment:
From what I understand of this, if you provide `--sign` via the CLI, it then will take whatever is the value of `@options[:sign_cert_file]` as part of the chain, but that option is not validated.

Is OK for the gemspec to have a `cert_chain` with `[nil]` in it?

If not, then `--cert` must be required once `--sign` is used and thus, require validation.

Reply:
Yeah, currently you would have an exception thrown because the cert isn't valid. I'll provide an error message!

Reply:
Done!